psycopg/psycopg/cursor.py

```diff
@@ -396,9 +426,10 @@ def _execute_results(
         if not results:
             raise e.InternalError("got no result from the query")
 
-        for res in results:
-            if res.status not in self._status_ok:
-                self._raise_from_results(results)
+        if not self._pgconn.pipeline_status:
```

Comment:
Maybe we can set a `results = ()` in the `if pipeline_mode` branch above to avoid this `if` here. Not for some mircro-optimisation but because this method is pretty complicated.

I still don't have a holistic view over the pipeline changes so I don't know if this will be changed later in the changeset I haven't seen yet, or if it can be simplified somehow.

Reply:
(ah just noticed this is the last changeset...)